File: python/changer.py
""" changeMaker.py
    get a price from user
    and an amount paid.
    determine how much change to give
    break change into twenties, tens ... etc

"""

#get a price from the user
price = raw_input("please tell me the price of the item: ")

#get an amount paid from the user
paid = raw_input("please tell me how much the user paid: ")

#convert all values to pennies
price = int(float(price) * 100)
paid = int(float(paid) * 100)

print "price: %d" % price
print "paid: %d" % paid

change = paid - price

print "Change due: %d " % change

twenties = change / 2000
print "twenties: %d" % twenties
change = change % 2000

tens = change / 1000
print "tens: %d" % tens
change = change % 1000

fives = change / 500
print "fives: %d" % fives
change = change % 500

ones = change / 100
print "ones: %d" % ones
change = change % 100

quarters = change / 25
print "quarters: %d" % quarters
change = change % 25

dimes = change / 10
print "dimes: %d" % dimes
change = change % 10

nickels = change / 5
print "nickels: %d" % nickels
change = change % 5

print "pennies: %d" % change

File: changer/Changer.java
// Changer.java
// Basic Changemaker lab

import java.util.*;

public class Changer{
  public static void main(String[] args){
    new Changer();
  } // end mian

  public Changer(){
    Scanner input = new Scanner(System.in);
    float itemCost;
    float tendered;
    float changeDue;
    int penniesLeft;

    System.out.println("How much does the item cost? ");
    itemCost = input.nextFloat();

    System.out.println("How much was paid? ");
    tendered = input.nextFloat();

    changeDue = tendered - itemCost;
    System.out.println("Change Due: " + changeDue);

    penniesLeft = (int)(tendered * 100) - (int)(itemCost * 100);
    System.out.println("PenniesLeft: " + penniesLeft);

    int twenties = (int)(penniesLeft / 2000);
    penniesLeft = penniesLeft % 2000;
    System.out.println("Twenties: " + twenties);

    int tens = (int)(penniesLeft / 1000);
    penniesLeft = penniesLeft % 1000;
    System.out.println("Tens: " + tens);
    
    int fives = (int)(penniesLeft / 500);
    penniesLeft = penniesLeft % 500;
    System.out.println("Fives: " + fives);

    int ones = (int)(penniesLeft / 100);
    penniesLeft = penniesLeft % 100;
    System.out.println("Ones: " + ones);

    int quarters = (int)(penniesLeft / 25);
    penniesLeft = penniesLeft % 25;
    System.out.println("Quarters: " + quarters);

    int dimes = (int)(penniesLeft / 10);
    penniesLeft = penniesLeft % 10;
    System.out.println("Dimes: " + dimes);

    int nickels = (int)(penniesLeft / 5);
    penniesLeft = penniesLeft % 5;
    System.out.println("Nickels: " + nickels);

    System.out.println("Pennies: " + penniesLeft);

  } // end constructor
} // end class def


    
